### PR TITLE
Add route for admins to issue tickets to users

### PR DESCRIPTION
**Pull request by aviupadhyayula**
Adds route for club admins to issue tickets to users. Tickets must already be created to be issued.

### CODE REVIEW

**Comment by aviupadhyayula** [comment]
This route bypasses `event.ticket_drop_time`, so tickets that aren't publicly available yet (i.e. through `event/tickets`) can still be issued. I think this fits the spirit of why we added the drop time. 

Thoughts?

File: backend/clubs/views.py [hunk 1]
```diff
@@ -2790,6 +2790,145 @@ def create_tickets(self, request, *args, **kwargs):
...
+                                detail:
+                                    type: string
+            "400":
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                detail:
+                                    type: string
```
Comment (joyliu-q) [inline backend/clubs/views.py:2832]: Thank you for this PR! Could the 400 detail also return an array of failed pennid strings?

Comment (aviupadhyayula) [reply backend/clubs/views.py:2832]: Sure, added an `errors` attribute to the response schema.

File: backend/clubs/views.py [hunk 2]
```diff
@@ -2790,6 +2790,165 @@ def create_tickets(self, request, *args, **kwargs):
 
         return Response({"detail": "Successfully created tickets"})
 
+    @action(detail=True, methods=["post"])
+    @transaction.atomic
+    @update_holds
+    def issue_tickets(self, request, *args, **kwargs):
```
Comment (rohangpta) [inline backend/clubs/views.py:2796]: Please lock behind same perms for `create_tickets`.

File: backend/clubs/views.py [hunk 3]
```diff
@@ -2790,6 +2790,165 @@ def create_tickets(self, request, *args, **kwargs):
...
+                                    type: string
+                                errors:
+                                    type: array
+                                    items:
+                                        type: string
+        ---
+        """
+        event = self.get_object()
+
+        quantities = request.data.get("quantities", [])
```
Comment (rohangpta) [inline backend/clubs/views.py:2841]: Nit: Is quantities an apt name here? We don't really have a count sub field

File: backend/clubs/views.py [hunk 4]
```diff
@@ -2790,6 +2790,165 @@ def create_tickets(self, request, *args, **kwargs):
...
+            ticket = next(
+                ticket
+                for ticket in tickets
+                if ticket.type == ticket_type and ticket.owner is None
+            )
+            ticket.owner = user
+            ticket.holder = None
+
+            transaction_records.append(
+                TicketTransactionRecord(
```
Comment (rohangpta) [inline backend/clubs/views.py:2931]: Would it be nice to have a "note: manually assigned here"? A null reconciliation id and 0.0 amount does tell us something was up...

File: backend/clubs/views.py [hunk 5]
```diff
@@ -2790,6 +2790,165 @@ def create_tickets(self, request, *args, **kwargs):
...
+                        ),
+                        "errors": [],
+                    },
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            tickets.extend(available_tickets)
+
+        # Hold all selected tickets for 10 mins
+        holding_expiration = timezone.now() + datetime.timedelta(minutes=10)
```
Comment (rohangpta) [inline backend/clubs/views.py:2912]: Why do we hold here?

**Review by rohangpta — COMMENTED** [review]
Cleanly written and well done. A few things to iron out but this is looking good to me overall. 

Permissions (and a test validating this) are the most important thing about this PR

File: backend/clubs/views.py (hunk 5 above)
Comment (aviupadhyayula) [reply backend/clubs/views.py:2912]: Forgot that this was wrapped with a `transaction.atomic`, good catch.

File: backend/clubs/views.py (hunk 3 above)
Comment (aviupadhyayula) [reply backend/clubs/views.py]: changed it to `tickets`

File: backend/clubs/views.py (hunk 4 above)
Comment (aviupadhyayula) [reply backend/clubs/views.py:2925]: I don't think it's necessary, but might be helpful (especially with the current impl of free tickets in #658, which also uses a null reconciliation ID & 0.0 amount). Happy to add a field to the model for comments. Can also hack the reconciliation ID, and set it to `issued` instead?

File: backend/clubs/views.py (hunk 2 above)
Comment (aviupadhyayula) [reply backend/clubs/views.py:2796]: thanks for catching

**Comment by aviupadhyayula** [comment]
> Permissions (and a test validating this) are the most important thing about this PR

Merging this in so that frontend dev isn't blocked, but main points should be addressed.

**Pull request merged**